transport_versioned_docs/version-1.0.0/workflow-client-server-secure.md

```diff
@@ -0,0 +1,147 @@
+Start by enabling Unity TLS with : 
+```cs 
+#define ENABLE_MANAGED_UNITYTLS
+```
+and add the following dependencies: 
+```cs
+using Unity.Collections;
+using Unity.Networking.Transport.TLS;
+```
+Then declare the secureParameters class and the boilerplate code that will hold your secure information:
+```cs
+public static class SecureParameters
+{
+        public static FixedString32Bytes commonName = new FixedString32Bytes("common-name");  // Use the common name you used to define the server certificate. 
+        public static FixedString4096Bytes myGameClientCA = new FixedString4096Bytes(
+        @"-----BEGIN CERTIFICATE-----
+                     ---                  
+          -----END CERTIFICATE-----"); // This should contain the content of myGameClientCA.pem 
+        public static FixedString4096Bytes myGameServerCertificate = new FixedString4096Bytes(
+        @"-----BEGIN CERTIFICATE-----
+                    ---
+          -----END CERTIFICATE-----"); // This should contain the content of myGameServerCertificate.pem  
+        public static FixedString4096Bytes myGameServerPrivate = new FixedString4096Bytes(
+        @"-----BEGIN RSA PRIVATE KEY-----
+                       ---
+          -----END RSA PRIVATE KEY-----"); // This should contain the content of myGameServerPrivate.key  
+}
+```  
+
+### Creating the Secure Server
+
+
+Starting from the minimal server sample code, create a `NetworkSettings` object.
+
+```cs
+private NetworkSettings settings = new NetworkSettings();
+```
+
+Within the `start()` method, configure this `NetworkSettings` as following: 
+```cs
+void Start ()
+    {
+        settings.WithSecureParameters(
+            hostname: ref SecureParameters.commonName,                    // Use the common name you used to generate `myGameServerCertificate.crt`
+            sslHandshakeTimeoutMin: 500,    // Duration in ms
+            sslHandshakeTimeoutMax: 2000,   // Duration in ms
+            pem:null,
+            rsa: ref SecureParameters.myGameServerCertificate,      // The content of the `myGameServerCertificate.crt`           
+            rsaKey: ref SecureParameters.myGameServerPrivate // The content of `myGameServerPrivate.key`
+        );
+```
+Call create method of the network driver but this time, integrate the `NetworkSettings` object.
+```cs 
+m_Driver = NetworkDriver.Create(settings); 
+```
+That's it for the server !
+
+### Creating a Secure Client
+
+The secure client is very similar to the secure server. Starting from the minimal client sample code, create `NetworkSettings` object.
+```cs
+private NetworkSettings settings = new NetworkSettings();
+```
+
+The difference between client and server lies with the parameters that will be provided to the NetworkSettings object.
+Within the `start()` method, configure this `NetworkSettings` object as following : 
+```cs
+void Start ()
+    {
+        settings.WithSecureParameters(
+            hostname: null,       
```

Comment:
I think it would make more sense to provide a version of `WithSecureParameters` without this argument. Also I'm not entirely sure `null` is valid here.

Reply:
Used the WithSecureServerParameters instead. I now omit the hostname.